### PR TITLE
DRY on password management; move eng sec policy to eng sec page

### PR DESCRIPTION
**Pull request by openprivacy**
----
:books: Documentation preview :books:: https://civicactions-handbook--1298.org.readthedocs.build/en/1298/

### CODE REVIEW

**Comment by openprivacy** [comment]
Password Policy is more cleanly separated from Password Management, and SSH and Server management policies have been moved into Engineering > [Security and Compliance](https://guidebook.civicactions.com/en/latest/practice-areas/engineering/security-compliance) -- this latter page needs some Engineering attention.

**Comment by openprivacy** [comment]
https://guidebook.civicactions.com/en/latest/practice-areas/engineering/security-compliance/ needs work, and this PR makes it [even more unwieldy](https://civicactions-handbook--1298.org.readthedocs.build/en/1298/practice-areas/engineering/security-compliance/); the next step will be to create a branch/PR to clean up this page.

[update] see #1299

**Comment by openprivacy** [comment]
Overview of Security Policy section changes:

1. **Password managers and two-factor authentication**
   - unique text integrated into common-practices-tools/security/README.md -- duplicate text removed
3. **Some password exceptions**
   - removed (if you want this, consider a move to practice-areas/engineering/security-compliance.md)
5. **Private keys**
   - moved to practice-areas/engineering/security-compliance.md
7. **Server and site security**
   - moved to practice-areas/engineering/security-compliance.md

The removal of sections that are only of interest to engineers should not require that everyone re-read/re-sign acceptance of the policy.

This was basically a tightening and clean-up of duplicated text. Next step will be to go over the [practice-areas/engineering/security-compliance.md](https://civicactions-handbook--1298.org.readthedocs.build/en/1298/practice-areas/engineering/security-compliance/) doc, tighten it up and ensure it contains policies that are general to all our engineers and devops teams. Tia.

File: practice-areas/engineering/security-compliance.md [hunk 1]
```diff
@@ -55,17 +55,92 @@ There are some instructions at <https://support.google.com/accounts/answer/18583
...
+-   Test the site to ensure settings are correct after changing settings affecting content/data access control.
+-   Avoid the use of PHP in the web administration interface when at all possible (as this code is harder to find and hence audit).
+-   Respect the privacy of site users, avoiding accessing personal data such as private messages.
+
+Developers and themers working on the site codebase (and committing code to Git) must also:
+
+-   Ensure their own code and development practices follow accepted secure coding standard.
+-   Ensure the standard dev-qa-live process is always followed, such that all changes that may affect site security can be thoroughly tested before being made live.
+-   Ensure that external developers (client or 3rd party) working on the site codebase are either:
+    -   A full part of our developer team, such that they been assessed/trained to have the appropriate skills and are subject to TL code review.
```
Comment (sfmcgee) [inline practice-areas/engineering/security-compliance.md]: I know this just moved from another doc, but we should define what 'TL code review' means somewhere.

File: practice-areas/engineering/security-compliance.md [hunk 2]
```diff
@@ -55,17 +55,92 @@ There are some instructions at <https://support.google.com/accounts/answer/18583
...
+
+IT team system administrators working on CivicActions servers must also:
+
+-   Take the utmost caution when working on server configuration - document and test each change.
+-   Non-urgent yet risky changes (those with significant risk of introducing undesired side-effects) should only be made when the person expects to remain online and available for a while after the change.
+-   Not work on site/user files as root - but "su" to the account first.
+-   Respect the privacy of server users, avoiding accessing others' personal data such as e-mails.
+-   Work with the IT team to ensure server and backup health is monitored and alerts are responded to promptly.
+-   Ensure offsite backups are transferred and stored only in encrypted form.
+-   Ensure the Hurricane Electric and RimuHosting access list (that controls remote hands and physical server access) is maintained.
```
Comment (sfmcgee) [inline practice-areas/engineering/security-compliance.md]: Is Hurricane Electric still in use?

File: practice-areas/engineering/security-compliance.md [hunk 3]
```diff
@@ -55,17 +55,92 @@ There are some instructions at <https://support.google.com/accounts/answer/18583
...
+
 ## Continuous Monitoring
 
 We use tools to support continuous monitoring for performance and efficiency, and to ensure proper operation and security. These tools include (not an exhaustive list):
 
 -   Event and error log capture: auditd (SELinux), fail2ban and AIDE.
 -   Continuous monitoring dashboards: LogWatch, LogStash/GrayLog, Nagios, Cloudwatch, Pingdom, StatusCake, OpsGenie and New Relic.
--   Automated security scanning: [OpenSCAP](http://www.open-scap.org/), [ansible-role-800-53](https://github.com/RedHatGov/ansible-role-800-53), and [GovReady](https://github.com/GovReady).
+-   Automated security scanning: OpenSCAP, OWASP ZAP, Trivy and Syft (for SBOM creation).
```
Comment (sfmcgee) [inline practice-areas/engineering/security-compliance.md]: Recommend adding Dependency Track for the SBOM portion of the list. Maybe consider a separate line for SBOM generation and tracking as it's kind of, but not really security scanning?

File: practice-areas/engineering/security-compliance.md (hunk 1 above)
Comment (openprivacy) [reply practice-areas/engineering/security-compliance.md]: Changed "TL" to "CivicActions"

File: practice-areas/engineering/security-compliance.md (hunk 2 above)
Comment (openprivacy) [reply practice-areas/engineering/security-compliance.md]: Removed "Hurricane Electric and "

File: practice-areas/engineering/security-compliance.md (hunk 3 above)
Comment (openprivacy) [reply practice-areas/engineering/security-compliance.md]: Added: "Supply chain and Software Bill of Materials (SBOM): Syft and DependencyTrack."

**Review by sfmcgee — APPROVED** [review]

**Review by dmundra — APPROVED** [review]

**Review by marctjones — APPROVED** [review]

**Pull request merged**